src/librustc_mir/const_eval/error.rs

```diff
@@ -50,7 +50,7 @@ impl Error for ConstEvalErrKind {}
 /// Turn an interpreter error into something to report to the user.
 /// As a side-effect, if RUSTC_CTFE_BACKTRACE is set, this prints the backtrace.
 /// Should be called only if the error is actually going to to be reported!
-pub fn error_to_const_error<'mir, 'tcx, M: Machine<'mir, 'tcx>>(
+pub fn error_to_const_error<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>>(
```

Comment:
This is a curious change, is this *needed* or just done for consistency?

Reply:
I see plenty of these bounds got added all over the place. The bound seems reasonable, I am just curious why this is happening now.

Reply:
The constraint comes from the stack frame [definition](https://github.com/rust-lang/rust/blob/96d77f0e5f103612d62b85938aacfb33f5768433/src/librustc_mir/interpret/eval_context.rs#L57):
```rust
pub body: &'mir mir::Body<'tcx>,
```
Since before the stack was part of the interpreter, all uses could infer this bound. Moving the stack behind the `Machine` trait requires this bound to be explicitly stated everywhere. I have not found a way to avoid this.